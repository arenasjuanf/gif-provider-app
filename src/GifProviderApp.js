import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import Grid from './components/Grid';

export const GifProviderApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <span>
                Gif-Provider
            </span> 
            < AddCategory setCategories={ setCategories } />
            <hr />

            <ul>
                {
                    categories.map((cat, idx) => {
                        return <Grid 
                            key={ idx } 
                            category={cat} 
                        />
                    }) 
                }
            </ul>
        </>
    )

}