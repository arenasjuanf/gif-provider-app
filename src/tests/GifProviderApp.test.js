import { shallow } from 'enzyme';
import {GifProviderApp} from '../GifProviderApp';
describe("<GifProviderApp/> tests", () => {

    test("<GifProviderApp/> component should show successfully", () => {

        const wrapper = shallow(<GifProviderApp />);
        expect(wrapper).toMatchSnapshot();

    })

    test("it should to show catergories list", () => {

        const categories = ["spiderman", "naruto"];

        const wrapper = shallow(<GifProviderApp defaultCategories={ categories } />);
        expect(wrapper.find('Grid').length).toBe(categories.length);

    })


})