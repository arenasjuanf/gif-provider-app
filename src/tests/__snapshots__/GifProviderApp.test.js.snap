// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifProviderApp/> tests <GifProviderApp/> component should show successfully 1`] = `
<Fragment>
  <span>
    Gif-Provider
  </span>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;
